### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ik-web**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,30 @@
...
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
+      let tempEl = newArr[i];
+
+      if (tempEl === null || tempEl === undefined) {
+        tempEl = '';
+      }
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:21]: invert this condition, so all you do is
```suggestion
      if (tempEl !== null || tempEl !== undefined) {
        result += tempEl
      }
```

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,30 @@
...
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newArr = [...this];
+    let sep = separator;
+    let result = '';
+
+    if (sep === null) {
+      sep = 'null';
+    }
```
[the remark this answers is not in the document]
Comment (ik-web) [reply src/arrayMethodJoin.js:13]: Fixed

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const newArr = [...this];
```
[the remark this answers is not in the document]
Comment (ik-web) [reply src/arrayMethodJoin.js]: Fixed

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,8 +4,30 @@
...
+    let sep = separator;
+    let result = '';
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
```
[the remark this answers is not in the document]
Comment (ik-web) [reply src/arrayMethodJoin.js]: Fixed

File: src/arrayMethodJoin.js [hunk 5]
```diff
@@ -4,8 +4,30 @@
...
+      let tempEl = newArr[i];
```
[the remark this answers is not in the document]
Comment (ik-web) [reply src/arrayMethodJoin.js]: Fixed

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**